Blizzard.IO.RabbitMQ/NetqRabbitPublisher.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using Blizzard.IO.RabbitMQ.Entities;
+using Blizzard.IO.Core;
+using EasyNetQ;
+using EasyNetQ.Topology;
+
+namespace Blizzard.IO.RabbitMQ
+{
+    public class NetqRabbitPublisher<TData> : IRabbitPublisher<TData>, IPublisherWithMetadata<TData, RabbitMessageProperties>, IPublisher<TData>
+    {
+        private readonly IBus _netqBus;
+        private readonly ISerializer<TData> _serializer;
+        private readonly IExchange _destinationExchange;
+        private readonly IConverter<RabbitMessageProperties, MessageProperties> _rabbitMessagePropertiesToMessagePropertiesConverter;
+        private readonly string _routingKey;
+        private readonly bool _isAbstract;
+
+        public NetqRabbitPublisher(IBus netqBus, ISerializer<TData> serializer, RabbitExchange destinationExchange, bool isAbstract = false, string routingKey = "/")
+        {
+            _netqBus = netqBus;
+            _serializer = serializer;
+            _destinationExchange = _netqBus.Advanced.ExchangeDeclare(destinationExchange.Name.ToLower(),
```

Comment:
Why are you declaring this Exchange?
(Im not saying it is wrong but we need to decide if the publisher and the consumer will declare the queue and exchange they are using to make sure they exist or not, cause rozi did something else)

Reply:
I did it because it's how the easyNetQ wrapper that we have now works

Reply:
Anyway we need to talk about it and decide what it our approach

Reply:
f